compiler/rustc_lint/src/types.rs

```diff
@@ -873,13 +873,19 @@ impl<'tcx> LateLintPass<'tcx> for TypeLimits {
             )
         }
 
-        fn partialeq_binop(diag_item: Symbol) -> Option<hir::BinOpKind> {
-            if diag_item == sym::cmp_partialeq_eq {
-                Some(hir::BinOpKind::Eq)
-            } else if diag_item == sym::cmp_partialeq_ne {
-                Some(hir::BinOpKind::Ne)
-            } else {
-                None
+        fn diag_item_binop(diag_item: Symbol) -> Option<Option<hir::BinOpKind>> {
```

Comment:
`Option<Option<` is not super clear, could you define an `enum ComparisonOp { BinOp(BinOpKind), Other }` or something like that?

Reply:
Yeah, that's way better. Thanks.